openlibrary/plugins/openlibrary/js/SearchBar.js

```diff
@@ -0,0 +1,298 @@
+        this.inCollapsibleMode = false;
+    }
+
+    /**
+     * Converts an already processed query into a search url
+     * @param {String} q query that's ready to get passed to the search endpoint
+     * @param {Boolean} [json] whether to hit the JSON endpoint
+     * @param {Number} [limit] how many items to get
+     */
+    composeSearchUrl(q, json, limit) {
+        let url = this.facetEndpoint;
+        if (json) {
+            url += '.json';
+        }
+        url += `?q=${q}`;
+        if (limit) {
+            url += `&limit=${limit}`;
+        }
+        url += `&mode=${SearchUtils.mode.read()}`;
+        return url;
+    }
+
+    /**
+     * Prepare an unprocessed query for book searching
+     * @param {String} q
+     * @return {String}
+     */
+    static marshalBookSearchQuery(q) {
+        if (q && q.indexOf(':') == -1 && q.indexOf('"') == -1) {
+            q = `title: "${q}"`;
+        }
+        return q;
+    }
+
+    /** Setup event listeners for autocompletion */
+    initAutocompletionLogic() {
+        // searches should be cancelled if you click anywhere in the page
+        $(document.body).on('click', this.clearAutocompletionResults.bind(this));
+        // but clicking search input should not empty search results.
+        this.$input.on('click', false);
+
+        this.$input.on('keyup', debounce(event => {
+            // ignore directional keys and enter for callback
+            if (![13,37,38,39,40].includes(event.keyCode)) {
+                this.renderAutocompletionResults();
+            }
+        }, 500, false));
+
+        this.$input.on('focus', debounce(event => {
+            event.stopPropagation();
+            this.renderAutocompletionResults();
+        }, 300, false));
+    }
+
+    /** Cleans up and performs the query, then update the autocomplete results */
+    renderAutocompletionResults() {
+        let q = this.$input.val();
+        if (q === '' || !(this.facetEndpoint in RENDER_AUTOCOMPLETE_RESULT)) {
+            return;
+        }
+        if (this.facet.read() === 'title') {
+            q = SearchBar.marshalBookSearchQuery(q);
+        }
+
+        this.$results.css('opacity', 0.5);
+        $.getJSON(this.composeSearchUrl(q, true, 10), data => {
+            const renderer = RENDER_AUTOCOMPLETE_RESULT[this.facetEndpoint];
+            this.$results.css('opacity', 1);
+            this.clearAutocompletionResults();
+            for (let d in data.docs) {
+                this.$results.append(renderer(data.docs[d]));
+            }
+        });
+    }
+
+    clearAutocompletionResults() {
+        this.$results.empty();
+    }
+
+    /**
+     * Updates the UI to match after the facet is changed
+     * @param {String} newFacet
+     */
+    handleFacetValueChange(newFacet) {
+        // update the UI
+        this.$facetSelect.val(newFacet);
+        const text = this.$facetSelect.find('option:selected').text();
+        $('header#header-bar .search-facet-value').html(text);
```

Comment:
Is this inside the class or outside? If outside move it outside. If inside, use a local selector defined in the constructor e.g. this.$facetSelect.find

Reply:
This will be fixed in an upcoming PR, so WONTFIX for now.

Reply:
I think it should be fixed in this PR :)